src/main/java/events/PurgeUserEvent.java

```diff
@@ -0,0 +1,88 @@
+ * You may obtain a copy of the License at
+
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * [START retail_purge_user_event]
+ * Import user events into a catalog from inline source using Retail API
+ */
+
+package events;
+
+import static events.setup.SetupCleanup.writeUserEvent;
+
+import com.google.api.gax.longrunning.OperationFuture;
+import com.google.cloud.retail.v2.PurgeMetadata;
+import com.google.cloud.retail.v2.PurgeUserEventsRequest;
+import com.google.cloud.retail.v2.PurgeUserEventsResponse;
+import com.google.cloud.retail.v2.UserEventServiceClient;
+import com.google.cloud.retail.v2.UserEventServiceSettings;
+import java.io.IOException;
+import java.util.concurrent.ExecutionException;
+
+public class PurgeUserEvent {
+
+  private static final String PROJECT_NUMBER = System.getenv("PROJECT_NUMBER");
+
+  private static final String ENDPOINT = "retail.googleapis.com:443";
+
+  private static final String DEFAULT_CATALOG = String.format(
+      "projects/%s/locations/global/catalogs/default_catalog",
+      PROJECT_NUMBER);
+
+  private static final String VISITOR_ID = "test_visitor_id";
+
+  // get user events service client
+  private static UserEventServiceClient getUserEventsServiceClient()
+      throws IOException {
+    UserEventServiceSettings userEventServiceSettings = UserEventServiceSettings.newBuilder()
+        .setEndpoint(ENDPOINT)
+        .build();
+    return UserEventServiceClient.create(userEventServiceSettings);
+  }
+
+  // get purge user event request
+  private static PurgeUserEventsRequest getPurgeUserEventRequest() {
+    PurgeUserEventsRequest purgeUserEventsRequest = PurgeUserEventsRequest.newBuilder()
+        // TO CHECK ERROR HANDLING SET INVALID FILTER HERE:
+        .setFilter(String.format("visitorId=\"%s\"", VISITOR_ID))
+        .setParent(DEFAULT_CATALOG)
+        .setForce(true)
+        .build();
+
+    System.out.printf("Purge user events request: %s%n",
+        purgeUserEventsRequest);
+
+    return purgeUserEventsRequest;
+  }
+
+  // call the Retail API to purge user event
+  public static void callPurgeUserEvents()
+      throws IOException, ExecutionException, InterruptedException {
+    OperationFuture<PurgeUserEventsResponse, PurgeMetadata> purgeOperation
+        = getUserEventsServiceClient().purgeUserEventsAsync(
+        getPurgeUserEventRequest());
+
+    System.out.printf("The purge operation was started: %s%n",
+        purgeOperation.getName());
+  }
+
+  public static void main(String[] args)
+      throws IOException, ExecutionException, InterruptedException {
+    writeUserEvent(VISITOR_ID);
+
+    callPurgeUserEvents();
+  }
+}
```

Comment:
`Purge user events request: parent: "projects/945579214386/locations/global/catalogs/default_catalog"
filter: "visitorId=\"test_visitor_id\""
force: true

The purge operation was started: projects/945579214386/locations/global/catalogs/default_catalog/operations/purge-user-events-18229227904214637144
[WARNING] thread Thread[Gax-1,5,events.PurgeUserEvent] was interrupted but is still alive after waiting at least 14999msecs
[WARNING] thread Thread[Gax-1,5,events.PurgeUserEvent] will linger despite being asked to die via interruption
[WARNING] thread Thread[grpc-nio-worker-ELG-1-1,5,events.PurgeUserEvent] will linger despite being asked to die via interruption
[WARNING] thread Thread[grpc-default-executor-0,5,events.PurgeUserEvent] will linger despite being asked to die via interruption
[WARNING] thread Thread[grpc-nio-worker-ELG-1-2,5,events.PurgeUserEvent] will linger despite being asked to die via interruption
[WARNING] thread Thread[grpc-default-executor-1,5,events.PurgeUserEvent] will linger despite being asked to die via interruption
[WARNING] thread Thread[grpc-nio-worker-ELG-1-3,5,events.PurgeUserEvent] will linger despite being asked to die via interruption
[WARNING] thread Thread[grpc-default-executor-2,5,events.PurgeUserEvent] will linger despite being asked to die via interruption
[WARNING] thread Thread[grpc-nio-worker-ELG-1-4,5,events.PurgeUserEvent] will linger despite being asked to die via interruption
[WARNING] thread Thread[Gax-2,5,events.PurgeUserEvent] will linger despite being asked to die via interruption
[WARNING] thread Thread[grpc-nio-worker-ELG-1-5,5,events.PurgeUserEvent] will linger despite being asked to die via interruption
[WARNING] thread Thread[grpc-nio-worker-ELG-1-6,5,events.PurgeUserEvent] will linger despite being asked to die via interruption
[WARNING] thread Thread[grpc-nio-worker-ELG-1-7,5,events.PurgeUserEvent] will linger despite being asked to die via interruption
[WARNING] thread Thread[grpc-nio-worker-ELG-1-8,5,events.PurgeUserEvent] will linger despite being asked to die via interruption
[WARNING] thread Thread[Gax-3,5,events.PurgeUserEvent] will linger despite being asked to die via interruption
[WARNING] thread Thread[Gax-4,5,events.PurgeUserEvent] will linger despite being asked to die via interruption
[WARNING] thread Thread[Gax-5,5,events.PurgeUserEvent] will linger despite being asked to die via interruption
[WARNING] thread Thread[Gax-6,5,events.PurgeUserEvent] will linger despite being asked to die via interruption
[WARNING] thread Thread[Gax-7,5,events.PurgeUserEvent] will linger despite being asked to die via interruption
[WARNING] thread Thread[Gax-8,5,events.PurgeUserEvent] will linger despite being asked to die via interruption
[WARNING] NOTE: 19 thread(s) did not finish despite being asked to  via interruption. This is not a problem with exec:java, it is a problem with the running code. Although not serious, it should be remedied.
[WARNING] Couldn't destroy threadgroup org.codehaus.mojo.exec.ExecJavaMojo$IsolatedThreadGroup[name=events.PurgeUserEvent,maxpri=10]
java.lang.IllegalThreadStateException
        at java.lang.ThreadGroup.destroy(ThreadGroup.java:778)
        at org.codehaus.mojo.exec.ExecJavaMojo.execute(ExecJavaMojo.java:293)
        at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:134)
        at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:207)
        at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:153)
        at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:145)
        at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:116)
        at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:80)
        at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build(SingleThreadedBuilder.java:51)
        at org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:128)
        at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:307)
        at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:193)
        at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:106)
        at org.apache.maven.cli.MavenCli.execute(MavenCli.java:863)
        at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:288)
        at org.apache.maven.cli.MavenCli.main(MavenCli.java:199)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:289)
        at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:229)
        at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:415)
        at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:356)
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 20.071 s
[INFO] Finished at: 2022-01-04T17:12:44+01:00
[INFO] Final Memory: 40M/726M
`